src/libraries/System.Private.Xml/src/System/Xml/Core/XmlEventCache.cs

```diff
@@ -432,7 +432,7 @@ public override void Flush()
         /// </summary>
         public override void WriteValue(object? value)
```

Comment:
I'll double check but I'm almost sure all implementations except this one are fine with null so IMO UntypedTypeConverter should work with nulls (especially other converter with similar structure is). This NullRef is especially very hard to hit (if even possible) by user since this XmlWriter is internal

Reply:
Actually you're right, seems I accidentally marked XmlRawTextWriter.WriteValue(object) (base class) as nullable even though it wasn't, marking as resolved (pending change)